agent/taskresource/grpcclient/credentialsfetcherclient.go

```diff
@@ -90,6 +93,71 @@ func (c CredentialsFetcherClient) AddKerberosLease(ctx context.Context, credenti
 	return credentialsFetcherResponse, nil
 }
 
+// AddNonDomainJoinedKerberosLease() invokes credentials fetcher daemon running on the host
+// to create kerberos tickets associated with gMSA accounts in domainless mode
+func (c CredentialsFetcherClient) AddNonDomainJoinedKerberosLease(ctx context.Context, credentialspecs []string, username string, password string, domain string) (CredentialsFetcherResponse, error) {
+	if len(credentialspecs) == 0 {
+		seelog.Error("credentialspecs request should not be empty")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	if len(username) == 0 || len(password) == 0 || len(domain) == 0 {
+		seelog.Error("username, password or domain should not be empty")
+		return CredentialsFetcherResponse{}, nil
+	}
+
+	defer c.conn.Close()
```

Comment:
Question - Since we are closing the connection in this method, the consumer of `CredentialsFetcherClient` needs to open the connection again before calling another method that uses the connection?

Reply:
yes, you are right, before every call connection needs to be established again